server/src/main/java/org/opensearch/cluster/routing/allocation/decider/RemoteStoreMigrationAllocationDecider.java

```diff
@@ -150,20 +154,41 @@ private Decision replicaShardDecision(ShardRouting replicaShardRouting, Discover
                 return allocation.decision(
                     Decision.NO,
                     NAME,
-                    getDecisionDetails(false, replicaShardRouting, targetNode, " since primary shard copy is not yet migrated to remote")
+                    getDecisionDetails(
+                        false,
+                        replicaShardRouting,
+                        targetNode,
+                        " since primary shard copy is not yet migrated to remote",
+                        allocation
+                    )
                 );
             }
             return allocation.decision(
                 Decision.YES,
                 NAME,
-                getDecisionDetails(true, replicaShardRouting, targetNode, " since primary shard copy has been migrated to remote")
+                getDecisionDetails(
+                    true,
+                    replicaShardRouting,
+                    targetNode,
+                    " since primary shard copy has been migrated to remote",
+                    allocation
+                )
             );
         }
-        return allocation.decision(Decision.YES, NAME, getDecisionDetails(true, replicaShardRouting, targetNode, ""));
+        return allocation.decision(Decision.YES, NAME, getDecisionDetails(true, replicaShardRouting, targetNode, "", allocation));
     }
 
     // get detailed reason for the decision
-    private String getDecisionDetails(boolean isYes, ShardRouting shardRouting, DiscoveryNode targetNode, String reason) {
+    private String getDecisionDetails(
+        boolean isYes,
+        ShardRouting shardRouting,
+        DiscoveryNode targetNode,
+        String reason,
+        RoutingAllocation allocation
+    ) {
+        if (allocation.debugDecision() == false) {
```

Comment:
```suggestion
        if (!allocation.debugDecision()) {
```

Reply:
Usually, in this repo I have seen `==false` getting used instead of `!` and hence used the same

Reply:
This is a convention we have been following for better readability .